src/vision/Vision.cpp

```diff
@@ -37,103 +65,107 @@ void leftCamCallback(const wfov_camera_msgs::WFOVImage::ConstPtr& msg)
-        int largestIndex = 0;
+        //Sort contours by size
+        std::sort(contours.begin(), contours.end(), compareContourAreas);
 
-        //Compare the first contour to other areas to find contour with the
-        //largest area
-        for (unsigned int i = 1; i < contours.size(); i++)
+        for (int i = 0; i < nLargest &&
+                        static_cast<unsigned int>(i) < contours.size(); ++i)
         {
-            double area = contourArea(contours[i], false);
-            if (area > largestArea)
+            //Find the moments (physical properties) of the contour
+            Moments moment;
+            moment = moments(contours[i], false);
+
+            int cx = -1;
+            int cy = -1;
+            int imWidth = msg->width;
+            int imHeight = msg->height;
+            //Using moments, find the center point of the contour
+            if (moment.m00 > 0)
             {
-                largestArea = area;
-                largestIndex = i;
+                //Find x and y coordinates with 0,0 being top left corner
+                cx = static_cast<int>(moment.m10/moment.m00);
+                cy = static_cast<int>(moment.m01/moment.m00);
+                if (doImShow)
+                {
+                    Point2f center = cv::Point2f(cx, cy);
+                    ROS_INFO_STREAM("Center at: " << "[" << cx - (imWidth/2) <<
+                                 "," << -1*(cy-(imHeight / 2)) << "]");
+                    circle(original, center, 5, Scalar(255, 255, 255), -1);
+                    //Draw a circle on the original image for location
+                    //visualization
+                    circle(original, center, 4, Scalar(0, 0, 255), -1);
+                }
             }
-        }
-
-        //Find the moments (physical properties) of the contour
-        Moments moment;
-        moment = moments(contours[largestIndex], false);
 
-        int cx = -1;
-        int cy = -1;
-        int imWidth = msg->image.width;
-        int imHeight = msg->image.height;
-        //Using moments, find the center point of the contour
-        if (moment.m00 > 0)
-        {
-            //Find x and y coordinates with 0,0 being top left corner
-            cx = static_cast<int>(moment.m10/moment.m00);
-            cy = static_cast<int>(moment.m01/moment.m00);
-            Point2f center = cv::Point2f(cx, cy);
-            std::cout << "Center at: " << "[" << cx - (imWidth/2) << "," <<
-                                    -1*(cy-(imHeight / 2)) << "]" << std::endl;
-            circle(original, center, 5, Scalar(255, 255, 255), -1);
-            //Draw a circle on the original image for location visualization
-            circle(original, center, 4, Scalar(0, 0, 255), -1);
+            ROS_DEBUG_STREAM("Preparing output");
+            ///Create the output message
+            robosub::visionPos outMsg;
+
+            //Prepare the output message
+            outMsg.xPos = cx - (imWidth / 2);
```

Comment:
This is definitely something to think about going forward. If it is deemed useful for such things as the AI, I will update it.